### ISSUE TITLE
Failing with Status code 502 since Dec 4?

### ISSUE THREAD

**Issue by emoreau99** [karlwancl/YahooFinanceApi#61]
Hi

Anybody else experiencing a "Flurl.Http.FlurlHttpException : Call failed with status code 502 (Connection timed out): GET https://fc.yahoo.com" error since Dec. 4?

Anyone found a workaround?

**Comment by kamoco** [comment]
I am hitting the same too.

**Comment by emoreau99** [comment]
found a workaround that seems to work.

In the InitAsync method of the "internal static class YahooSession" class:

-on the line : var response = await "https://login.yahoo.com"
     -replace login with fc
-replace the line "_cookie = response.Cookies[0];"
    -with : _cookie = response.Cookies.FirstOrDefault(c => c.Name == "A3");

**Comment by kamoco** [comment]
Those changes did not work. The response here still throws 502.

   var response = await "https://fc.yahoo.com"
       .AllowHttpStatus("404")
       .WithHeader(userAgentKey, userAgentValue)
       .GetAsync()
       .ConfigureAwait(false);

**Comment by BobMc98** [comment]
The workaround from @emoreau99 worked well for me, with the following clarification:
> -replace login with fc

For me, I did the opposite. I had "fc.yahoo.com" on that line, and I replaced it with "login.yahoo.com". Everything worked then.

**Comment by kamoco** [comment]
hmm I tried both and none of them worked.

**Comment by BobMc98** [comment]
@kamoco My code now looks like this:

```       /// <exception cref="System.Exception">Failure to create client.</exception>
       public static async Task InitAsync(CancellationToken token = default)
       {
           if (_crumb != null)
           {
               return;
           }

           await _semaphore.WaitAsync(token).ConfigureAwait(false);
           try
           {
               const string userAgentKey = "User-Agent";
               const string userAgentValue = "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36";

               var response = await "https://login.yahoo.com"
                   .AllowHttpStatus("404")
                   .WithHeader(userAgentKey, userAgentValue)
                   .GetAsync()
                   .ConfigureAwait(false);

               if (response.Cookies.Count == 0)
               {
                   throw new Exception("Failed to obtain Yahoo auth cookie.");
               }
               else
               {
                   _cookie = response.Cookies.FirstOrDefault(c => c.Name == "A3");
```

**Comment by kamoco** [comment]
This worked for me: 

change login to fc
Allow 502
replace the _cookie line -  _cookie = response.Cookies.FirstOrDefault(c => c.Name == "A3");

 var response = await "https://fc.yahoo.com"
     .AllowHttpStatus("502")
     .WithHeader(userAgentKey, userAgentValue)
     .GetAsync()
     .ConfigureAwait(false);

 if (response.Cookies.Count == 0)
 {
     throw new Exception("Failed to obtain Yahoo auth cookie.");
 }
 else
 {
     //_cookie = response.Cookies[0];
     _cookie = response.Cookies.FirstOrDefault(c => c.Name == "A3");

**Comment by reviloxx** [comment]
For me the request to "https://fc.yahoo.com" runs into a timeout and if I try "https://login.yahoo.com" I only get a cookie with name "AS". With this cookie the next request (_crumb = await "https://query1.finance.yahoo.com/v1/test/getcrumb") throws an error.

**Comment by irigopou** [comment]
Same as above. Location in Germany, just in case this matters.

**Comment by sergemat** [comment]
Fixed with PRs #62 and #63.
